packages/kbn-test/src/es/test_es_cluster.ts

```diff
@@ -88,31 +151,51 @@ export function createTestEsCluster(options: TestClusterFactoryOptions) {
     }
 
     async start() {
-      let installPath;
+      let installPath: string;
 
+      // We only install once using the first node. If the cluster has
+      // multiple nodes, they'll all share the same ESinstallation.
+      const firstNode = this.nodes[0];
       if (esFrom === 'source') {
-        installPath = (await cluster.installSource(config)).installPath;
+        installPath = (await firstNode.installSource(config)).installPath;
       } else if (esFrom === 'snapshot') {
-        installPath = (await cluster.installSnapshot(config)).installPath;
+        installPath = (await firstNode.installSnapshot(config)).installPath;
       } else if (Path.isAbsolute(esFrom)) {
         installPath = esFrom;
       } else {
         throw new Error(`unknown option esFrom "${esFrom}"`);
       }
 
-      if (dataArchive) {
-        await cluster.extractDataDirectory(installPath, dataArchive);
-      }
+      for (let i = 0; i < this.nodes.length; i++) {
```

Comment:
can we run this logic in parallel to speed up the process?

Reply:
Good idea, yeah we should be able to parallelize the data directory extraction & starting nodes at least.

Same can be done for stopping nodes.

Reply:
Ah actually, parallelizing starting up the nodes is going to be problematic based on how the native realm setup is currently handled (where we wait to do it only after the last node is started).

I can start with `extractDataDirectory` and `stop` though.

Reply:
Updated